pkg/build/webhook/webhook.go

```diff
@@ -1,34 +1,65 @@
 package webhook
 
 import (
-	"fmt"
+	"crypto/hmac"
+	"errors"
 	"strings"
 
 	"github.com/openshift/origin/pkg/build/api"
 )
 
+const (
+	refPrefix = "refs/heads/"
+)
+
 var (
-	ErrSecretMismatch = fmt.Errorf("the provided secret does not match")
-	ErrHookNotEnabled = fmt.Errorf("the specified hook is not enabled")
+	ErrSecretMismatch = errors.New("the provided secret does not match")
+	ErrHookNotEnabled = errors.New("the specified hook is not enabled")
 )
 
 // GitRefMatches determines if the ref from a webhook event matches a build configuration
-func GitRefMatches(eventRef, configRef string) bool {
-	const RefPrefix = "refs/heads/"
-	eventRef = strings.TrimPrefix(eventRef, RefPrefix)
-	configRef = strings.TrimPrefix(configRef, RefPrefix)
-	if configRef == "" {
+func GitRefMatches(eventRef string, buildSource *api.BuildSource) bool {
+	var configRef string
+	if buildSource.Git == nil || buildSource.Git.Ref == "" {
 		configRef = "master"
+	} else {
+		configRef = buildSource.Git.Ref
 	}
```

Comment:
for bonus points you can put the TrimPrefix inside the if as well so it only gets called when you're using a a value of unknown origin (no point in trimming "master")